### ISSUE TITLE
NoneType object has no attribute select_folder

### ISSUE THREAD

**Issue by reportaman** [soywod/iris.vim#43]
When I run ``:Iris`` & enter my password for imap,smtp, it fails with the following message:

```
Iris: logging in...
Iris: selecting folder...
Iris: job: connection lost
Iris: api: 'open() takes 3 positional arguments but 4 were given'
Iris: api: '''NoneType'' object has no attribute ''select_folder'''
```

**Relevant information**
```
System: macOS BigSur 11.2 Apple Silicon/ M1 ARM architecture

Vim:
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Dec 28 2020 07:15:57)
macOS version
Included patches: 1-2164
Compiled by Homebrew
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               +file_in_path      -mouse_sysmouse    -tag_old_static
+arabic            +find_in_path      +mouse_urxvt       -tag_any_white
+autocmd           +float             +mouse_xterm       +tcl
+autochdir         +folding           +multi_byte        +termguicolors
-autoservername    -footer            +multi_lang        +terminal
+balloon_eval      +fork()            -mzscheme          +terminfo
+balloon_eval_term +fullscreen        +netbeans_intg     +termresponse
+browse            +gettext           +num64             +textobjects
++builtin_terms    -hangul_input      +odbeditor         +textprop
+byte_offset       +iconv             +packages          +timers
+channel           +insert_expand     +path_extra        +title
+cindent           +ipv6              +perl              +toolbar
+clientserver      +job               +persistent_undo   +transparency
+clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3           +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       +lua               +ruby              +wildmenu
+dialog_con_gui    +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       -X11
+dnd               +mouse             -sound             -xfontset
-ebcdic            +mouseshape        +spell             +xim
+emacs_tags        +mouse_dec         +startuptime       -xpm
+eval              -mouse_gpm         +statusline        -xsmp
+ex_extra          -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
-farsi             +mouse_sgr         +tag_binary
```

**Comment by gt00x** [comment]
I had the same error message/problem.  In the "iris-api.log", I could see that this happens when the login was ok and it was selecting/reading the Inbox folder. When I then used ':IrisFolder' or gf  to select another folder, it worked for some folders. 

For me the issue started when I had one plain text test email on the top of my Inbox list. **I removed that email** and it made it work again. 

From the iris-api.log I could see that this message adds "None" for most fields that "Iris" processes for this email... and that make it fail. (This is my assumption, after it just started working again.)

(I like this project ... unfortunately there are still a few bugs to fix I hope Soywod has time again at some point ...  I am not a programmer)

Cheers
